### PR TITLE
Berkshelf support

### PR DESCRIPTION
**Pull request by kfalconer**
Since there is support for `metadata.rb` chef style version bumps. We are using this with chef cookbooks to bump versions. Chef has its own dependency resolution project called Berkshelf, works very similar to Bundler. This PR adds support for Berkshelf.

It will automatically execute the `berks` command when a `Berksfile.lock` file is detected.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1,14 +1,94 @@
 PATH
   remote: .
   specs:
-    bump (0.5.3)
+    bump (0.5.12)
```
Comment (grosser) [inline Gemfile.lock]: do not bump in branches ... please revert that version change

File: Gemfile.lock [hunk 2]
```diff
@@ -1,14 +1,94 @@
...
 
 GEM
   remote: https://rubygems.org/
   specs:
+    addressable (2.4.0)
+    berkshelf (4.3.5)
```
Comment (grosser) [inline Gemfile.lock]: so many dependencies :(

File: Gemfile.lock (hunk 1 above)
Comment (gregorym) [inline Gemfile.lock]: And rebase please.

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -1,4 +1,5 @@
 require "bundler"
+require 'berkshelf'
```
Comment (grosser) [inline spec/spec_helper.rb]: why is this needed ?

File: README.md [hunk 4]
```diff
@@ -40,6 +40,11 @@ If you don't want to run the `bundle` command after bumping, add the `--no-bundl
     
     bump patch --no-bundle
 
+### --no-berks
+If you don't want to run the `berks` command after bumping, add the `--no-berks` option.
```
Comment (gregorym) [inline README.md]: It should be the other way around. Pass a flag if you want to run `berks`

**Comment by grosser** [comment]
FYI instead of adding berkshelf a berks stub could be a simple/lightweight way of testing berks integration:

```Ruby
around do |t|
  in_tmp_dir do
    File.write('berks', '# /usr/bin/env/ruby\nFile.write('Berklsfile.lock', '1.0.0')')
    File.chmod(0775, 'berks')
    with_env(PATH: "#{ENV["PATH"]}:.") do
      test.call
    end
  end
end
```

**Comment by grosser** [comment]
build is broken ...

File: README.md (hunk 4 above)
Comment (grosser) [inline README.md]: bundler is automatic too ? should berks not be (if there is a Berksfile) ?

Comment (gregorym) [inline README.md]: yeah you're right we should keep it consistent.

File: Gemfile.lock (hunk 1 above)
Comment (kfalconer) [inline Gemfile.lock]: @grosser good catch, the test specs were executing against the project Gemfile.lock

File: Gemfile.lock (hunk 2 above)
Comment (kfalconer) [inline Gemfile.lock]: I'll see what I can do to mock out berkshelf in the tests to remove the dependencies. However these would only be development/test dependencies. Any project using bump would not be required to install this.

File: spec/spec_helper.rb (hunk 3 above)
Comment (kfalconer) [inline spec/spec_helper.rb]: You are correct, this is no longer needed. At one point I was checking for the existence of the constant `Berkshelf` Eg:

````ruby
      def bundler_with_clean_env(&block)
        if defined?(Bundler)
          Bundler.with_clean_env(&block)
        else
          yield
        end
      end
````

But this ended up not being needed. I should be able to remove this as a development/test dependency
